### PR TITLE
[CAS-1366] Redraw attachments only when there's a change in diff

### PR DESCRIPTION
**Pull request by leandroBorgesFerreira**
[CAS-1366](https://stream-io.atlassian.net/browse/CAS-1366)

### 🎯 Goal

Don't reload giphys when the attachments don't change

### 🎨 UI Changes

<table>
<thead>
<tr>
<th>Before</th>
<th>After</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<video src="https://user-images.githubusercontent.com/10619102/135907395-d8cd8615-1b9a-43b3-8145-dbdc53be2460.mov
" controls="controls" muted="muted" />
</td>
<td>
<video src="https://user-images.githubusercontent.com/10619102/135907277-f6cd65cf-12ce-49d2-8ca1-0e51597a2b08.mov
" controls="controls" muted="muted" />
</td>
</tr>
</tbody>
</table>

### 🧪 Testing

Make reaction in a giphy

### ☑️ Checklist

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] PR targets the `develop` branch
- [ ] Changelog is updated with client-facing changes. todo
- ~[ ] New code is covered by unit tests~
- [x] Comparison screenshots added for visual changes
- ~[ ] Affected documentation updated (docusaurus, tutorial, CMS (task created))~
- [x] Reviewers added

### 🎉 GIF

_Please provide a suitable gif that describes your work on this pull request_

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -65,12 +65,13 @@
 ## stream-chat-android-ui-components
 ### 🐞 Fixed
 - Fixed bug related to scroll of messages.
-
+- Updating attachments view holder only when att
```
Comment (adasiewiczr) [inline CHANGELOG.md:68]:
```suggestion
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -65,12 +65,13 @@
...
+- Updating attachments view holder only when attachments have changed. This fixes a problem with reloading gifs when reactions are added or removed in then.
```
Comment (adasiewiczr) [inline CHANGELOG.md:69]:
```suggestion
- Updating attachments view holder only when attachments have changed. This fixes a problem with reloading gifs when reactions are added or removed.
```

**Review by adasiewiczr — APPROVED** [review]
We need to review all view holders and make sure we are using diff payloads properly. I'll add a task for that

File: stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/viewholder/internal/TextAndAttachmentsViewHolder.kt [hunk 3]
```diff
@@ -75,7 +75,10 @@ internal class TextAndAttachmentsViewHolder(
         binding.messageText.isVisible = data.message.text.isNotEmpty()
         markdown.setText(binding.messageText, data.message.text)
 
-        setupAttachment(data)
+        if (diff?.attachments == true) {
```
Comment (ogkuzmin) [inline stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/viewholder/internal/TextAndAttachmentsViewHolder.kt:78]: Seems like if diff is null we don't draw attachments. Is it right? Maybe we can make diff not-nullable passing some default empty value for example

Comment (leandroBorgesFerreira) [reply stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/message/list/adapter/viewholder/internal/TextAndAttachmentsViewHolder.kt:78]: The default empty value wouldn't be the original/correct data, which can be confusing. Fixing the nullability of the diff is outside of the scope of this PR. Let's create another PR an investigate why the diff can be null. 

About the draw with null diff, what about drawing it if the diff is null and only not drawing when it is false? This is closer from the current behaviour

**Review by bychkovdmitry — APPROVED** [review]

**Review by ogkuzmin — APPROVED** [review]

**Pull request merged**